### PR TITLE
Packaging PR - Getting dbt-semantic-interfaces ready for PyPi

### LINKED ISSUE #27: [Feature] Publish dbt-semantic-interfaces on PyPi
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward change to existing dbt-semantic-interfaces functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

As soon as we've got a working version, we should publish `dbt-semantic-interfaces` on PyPi so that both dbt-core and MetricFlow can install it without needing to reference the github link.

### Describe alternatives you've considered

Having them reference the hard coded git link but that seems like a 🅱️ ad idea

### Who will this benefit?

The developers of MetricFlow & dbt-core!

### Are you interested in contributing this feature?

Yarp

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by callum-mcdata**
resolves #27 

### Description
This draft PR is not ready yet but it contains the working branch for getting DSI ready for PyPi.

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -34,6 +34,17 @@ dependencies = [
 requires = ["hatchling"]
 build-backend = "hatchling.build"
 
+[tool.hatch.build.targets.sdist]
+exclude = [
```
Comment (QMalcolm) [inline pyproject.toml:38]: we should also exclude `/tests` 🙂

**Review by QMalcolm — COMMENTED** [review]
One small tweak

File: pyproject.toml (hunk 1 above)
Comment (callum-mcdata) [reply pyproject.toml:38]: Updated!

**Review by QMalcolm — APPROVED** [review]
Thank you @callum-mcdata for this work

**Pull request merged**